Git Sparse Checkout

Srinivasan-tm opened the pull request:
Git Sparse checkout script for selectively cloning folders from a repository

tmsrinivasantm committed:
Sparse checkout script to selectively clone folders from a repo

Isha-Sharma commented on git/sparse_checkout.sh:
```diff
@@ -0,0 +1,27 @@
+
+test -d "$MAIN_DIR" || mkdir "$MAIN_DIR"
+cd "$MAIN_DIR" || exit
+test -d .git || git init --initial-branch=main
```
Can get branch from param? If not, default to main.

Srinivasan-tm replied:
I've made this change, can I resolve this conversation?

Isha-Sharma commented on git/sparse_checkout.sh:
```diff
@@ -0,0 +1,27 @@
+SUBDIR="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\4|p')"
+
+test -d "$MAIN_DIR" || mkdir "$MAIN_DIR"
+cd "$MAIN_DIR" || exit
```
Why do we need || exit?

Srinivasan-tm replied:
good practice, if cd fails instead of carrying out the rest of the operations, we just exit

Isha-Sharma replied:
okay, but since we're creating the folder in the line above, cd would never fail?

Srinivasan-tm replied:
true, its more of a habitual thing, I can remove it if necessary

Isha-Sharma commented on git/sparse_checkout.sh:
Add comments and logs

Srinivasan-tm replied:
I'll add comments, but logs I feel are already sufficient from the ones that the git commands provide by itself

Srinivasan-tm replied:
I've sent it on Slack

Commit:
Added branch argument for sparse_checkout

* Optionally include branch to pull from, defaults to "main"

Commit:
Added relevant comments, and modified help message

Commit:
Removed unnecessary failsafe

* cd || exit is a popular pattern to make sure your program exits if cd
fails, but there is no change cd fails in this scenario as we are
performing mkdir in the previous step, so removing "|| exit"

rickygarg commented on git/sparse_checkout.sh:
```diff
@@ -0,0 +1,37 @@
+if [ "$(git remote show | wc -l)" = "0" ]; then
+  git remote add origin "$REMOTE"
+fi
+git config core.sparseCheckout true
```
I don't want to make permanent changes to my config. Can we enable flag only on commands that need it.

rickygarg commented on git/sparse_checkout.sh:
```diff
@@ -0,0 +1,37 @@
+fi
+
+# This specifies the remote URL from which to fetch from
+REMOTE="$(echo "$URL" | sed -n 's|\(.*github.com\)/\([^/]*\)/\([^/]*\)/\(.*\)|\1/\2/\3.git|p')"
```
Can it be generic so that it works on non github.com URLs also

Commit:
Approaching sparse checkout in a different method

tmsrinivasantm committed:
Fixed issue where running this script on a sparse-checkout branch with
missing files did not restore the missing files + fixed issue where
running this script did not pull remote changes to local
